semantics/src/namespace/types.rs

```diff
@@ -93,12 +93,24 @@ pub enum FixedSize {
 
 #[derive(Clone, Debug, PartialEq, PartialOrd, Ord, Eq)]
 pub enum Base {
-    U256,
+    Numeric(Integer),
     Bool,
     Byte,
     Address,
 }
 
+#[derive(Clone, Debug, PartialEq, PartialOrd, Ord, Eq)]
+pub enum Integer {
+    U256,
+    U128,
+    U64,
+    U32,
+    U16,
+    U8,
+}
```

Comment:
My idea is to let the `Integer` enum host everyting from `u8`..`u256`  to `i8`..`i256`. This is to keep the hierarchy shorter and hence simplify matching etc.